src/overlays/actors/ovl_Obj_Etcetera/z_obj_etcetera.c

```diff
@@ -26,35 +32,286 @@ const ActorInit Obj_Etcetera_InitVars = {
+    DynaPolyActor* dyna = &this->dyna;
+    f32 scaleTemp;
+
+    if (func_800CAF70(dyna)) {
+        this->burrowFlag |= 1;
+    } else {
+        this->burrowFlag &= ~1;
+    }
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->oscillationTimer > 0) {
+        this->oscillationTimer--;
+    } else {
+        this->dyna.actor.draw = ObjEtcetera_DrawIdle;
+        this->actionFunc = ObjEtcetera_Idle;
+        Actor_SetScale(&this->dyna.actor, 0.01f);
+        this->dyna.actor.scale.y = 0.02f;
+        this->oscillationTimer = 0;
+        this->intenseOscillationScale = 0.0f;
+        return;
+    }
+    this->intenseOscillationScale *= 0.8f;
+    this->intenseOscillationScale -= (this->dyna.actor.scale.x - 0.01f) * 0.4f;
+    scaleTemp = dyna->actor.scale.x + this->intenseOscillationScale;
+    Actor_SetScale(&this->dyna.actor, scaleTemp);
+    this->dyna.actor.scale.y = 2.0f * scaleTemp;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Etcetera/ObjEtcetera_Destroy.s")
+void ObjEtcetera_Setup(ObjEtcetera* this, GlobalContext* globalCtx) {
+    CollisionHeader* colHeader = NULL;
+    ObjEtceteraType type;
+    CollisionHeader* collisionHeaders[] = { &D_0400E710, &D_0400E710, &D_040118D8, &D_040118D8 };
+    s32 pad;
+    CollisionHeader* thisCollisionHeader;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Etcetera/func_80A7BDC8.s")
+    type = OBJETCETERA_TYPE(&this->dyna.actor);
+    if ((type < TYPE_PINK_FLOWER) || (type >= NUMBER_OF_FLOWER_TYPES)) {
+        type = TYPE_PINK_FLOWER;
+    }
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objIndex)) {
+        this->dyna.actor.objBankIndex = this->objIndex;
+        Actor_SetObjectSegment(globalCtx, &this->dyna.actor);
+        BcCheck3_BgActorInit(&this->dyna, 1);
+        thisCollisionHeader = collisionHeaders[type];
+        if (thisCollisionHeader != 0) {
+            BgCheck_RelocateMeshHeader(thisCollisionHeader, &colHeader);
+        }
+        this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, colHeader);
+        type = OBJETCETERA_TYPE(&this->dyna.actor);
+        switch (type) {
+            case TYPE_PINK_FLOWER:
+            case TYPE_PINK_FLOWER_SPAWNED_FROM_MAD_SCRUB:
+                SkelAnime_Init(globalCtx, &this->skelAnime, &D_04011518, &D_0400EB7C, this->jointTable,
+                               this->morphTable, 11);
+                this->dList = &D_0400ED80;
+                break;
+            case TYPE_GOLD_FLOWER:
+            case TYPE_GOLD_FLOWER_SPAWNED_FROM_MAD_SCRUB:
+                this->dList = &D_04011BD0;
+                SkelAnime_Init(globalCtx, &this->skelAnime, &D_040127E8, &D_0400EB7C, this->jointTable,
+                               this->morphTable, 11);
+                this->collider.dim.height = 20;
+                break;
+        }
+        type = OBJETCETERA_TYPE(&this->dyna.actor);
+        switch (type) {
+            case TYPE_PINK_FLOWER:
```

Comment:
This function's quite packed together. I'd add some blank lines to make it easier to read.